examples/features/torch_batch_process/batch_inference/generate_embedding/bert_embedding_generation.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import pathlib
+
+import chromadb
+import torch
+from datasets import load_dataset
+from transformers import BertModel, BertTokenizer
+
+from determined.pytorch import experimental
+
+
+class EmbeddingProcessor(experimental.TorchBatchProcessor):
+    def __init__(self, context):
+        self.tokenizer = BertTokenizer.from_pretrained("bert-base-uncased")
+        self.model = BertModel.from_pretrained("bert-base-uncased", output_hidden_states=True)
+
+        self.device = context.device
+
+        self.model = context.prepare_model_for_inference(self.model)
+
+        self.output = []
+
+        self.context = context
+        self.last_index = 0
+        self.rank = self.context.distributed.get_rank()
+
+        self.output_dir = "/tmp/data/embeddings"
+        os.makedirs(self.output_dir, exist_ok=True)
+
+    def process_batch(self, batch, batch_idx) -> None:
+        with torch.no_grad():
+            tokenized_text = self.tokenizer.batch_encode_plus(
+                batch["text"],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+                add_special_tokens=True,
+            )
+            inputs = torch.tensor(tokenized_text["input_ids"])
+            inputs = inputs.to(self.device)
+            masks = torch.tensor(tokenized_text["attention_mask"])
+            masks = masks.to(self.device)
+
+            outputs = self.model(inputs, masks)
+            outputs = torch.mean(outputs["hidden_states"][-1], dim=1)
+
+            self.output.append({"embeddings": outputs, "id": batch["_id"], "text": batch["text"]})
+            self.last_index = batch_idx
+
+    def on_checkpoint_start(self):
+        if len(self.output) == 0:
```

Comment:
nit: can we add some explanation in `on_checkpoint_start` and `on_finish` to explain to the user what is happening in the code? I think that a novice user that wants to create a vector db, may not know immediately e.g., (1) why file name should contain rank or last index; (2) if the data is saved to a checkpoint; (3) that `on_finish` is where vector db is created after all documents have been processed.